src/frontend/packages/lib_components/src/common/Typo/index.tsx

```diff
@@ -0,0 +1,84 @@
+import {
+  CSSProperties,
+  DetailedHTMLFactory,
+  ReactHTML,
+  createElement,
+} from 'react';
+
+type Values<T> = T[keyof T];
+export type ObtainHTMLProps<T extends Values<ReactHTML>> =
+  T extends DetailedHTMLFactory<infer Props, HTMLElement> ? Props : never;
+
+export interface TypoPropsOnly<T> {
+  textAlign?: CSSProperties['textAlign'];
+  color?: CSSProperties['color'];
+  fontSize?: CSSProperties['fontSize'];
+  truncate?: boolean | number;
+  type?: keyof T;
+}
+
+export type TypoProps<T extends keyof ReactHTML> = TypoPropsOnly<ReactHTML> &
+  ObtainHTMLProps<ReactHTML[T]>;
```

Comment:
Can you add comment on this type please. They are not trivial.

Reply:
I simplified a bit the interfaces, plus added comments:
- [7f2063a](https://github.com/openfun/marsha/pull/2427/commits/7f2063a974e0a965757ed44cbb94003df25f6c4a)
- [5782f2f](https://github.com/openfun/marsha/pull/2427/commits/5782f2fe82b8ffcab1e6deb69fe45151b01fce7a)